kvdb-rocksdb/src/lib.rs

```diff
@@ -507,8 +507,13 @@ impl Database {
 							let end_range = end_prefix.unwrap_or_else(|| vec![u8::max_value(); 16]);
 							batch.delete_range_cf(cf, &prefix[..], &end_range[..]);
 							if no_end {
+								use crate::iter::IterationHandler as _;
+
 								let prefix = if prefix.len() > end_range.len() { &prefix[..] } else { &end_range[..] };
-								for (key, _) in self.iter_with_prefix(col, prefix) {
+								// We call `iter_with_prefix` directly on `cfs` to avoid taking a lock twice
+								// See https://github.com/paritytech/parity-common/pull/396.
+								let read_opts = generate_read_options();
```

Comment:
Why is it ok to not call `read_opts.set_iterate_upper_bound()` here?

Reply:
Oh :+1: I think I agree with you we should need the upper bound (iteration_handler function name 'iter_with_prefix'  is a bit misleading it is more 'from_prefix_with_option').

Reply:
> Why is it ok to not call `read_opts.set_iterate_upper_bound()` here?

Because here end_prefix is `None` (`no_end`).

Reply:
:facepalm: (monday are difficult, I remember telling to myself on first read that the code skipped one call to end_prefix thanks to that while I read that). So yes, this is actually correct (I mean there is no issue with this).

Reply:
> Because here end_prefix is `None` (`no_end`).

…because we're deleting all values in the column... got it!